### DISCUSSION TITLE
How to specify commit or tag for Menagerie models?

### DISCUSSION THREAD

**Discussion by kevinzakka** [robot-descriptions/robot_descriptions.py#86] [Q&A] [answered]
Hi Stéphane,

In Menagerie, we continually update models to improve them. This can sometimes bump up the minimum required MuJoCo version for that model. If a user does not want to upgrade and would rather stick to an older version, what is the recommended way to specify a commit or tag for Menagerie models in `robot_descriptions` without having to downgrade `robot_descriptions` itself?

**Comment by stephane-caron** [comment, marked as answer]
Sure thing: loaders in robot-descriptions have a ``commit`` keyword argument to pin the version of any robot description. For instance, to load a UR5e from a specific commit of MuJoCo Menagerie:

```py
from robot_descriptions.loaders.mujoco import load_robot_description

model = load_robot_description(
    "ur5e_mj_description",
    commit="cc3868ec21bae324e6dd3b01ea185b6e6f0fa172",
)
```

I just pushed this example in [specific_commit.py](https://github.com/robot-descriptions/robot_descriptions.py/blob/4a5c9fbb27107a0a55c2407402a94e41844342e3/examples/specific_commit.py).

**Reply by kevinzakka** [reply]
Sometimes I don't want to get an `MjModel` object, for instance when I want to load the model with dm_control's [PyMJCF](https://github.com/google-deepmind/dm_control/tree/main/dm_control/mjcf#pymjcf). We should probably look into returning the MJCF_PATH if requested by the user.

**Comment by kevinzakka** [comment]
I should have looked at the code more closely 🤦‍♂️ I have only been using the "imported module" API and forgot this was possible. Thanks @stephane-caron!

**Reply by stephane-caron** [reply]
No worries :ok_hand: Reopening the discussion as the question is relevant and can be useful to others discovering the library. (Contrary to issues, [closing discussions](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#closing-a-discussion) is mainly when they are not relevant or duplicate.)
